src/components/number/number.jsx

```diff
@@ -28,7 +28,11 @@ const Number = React.forwardRef((props, ref) => {
     // The MDS Number is using an uncontrolled `<input>`.
     // In order to set a new provided value prop, we
     // set the internal state of the `<input>`.
-    inputRef.current.value = props.value;
+    if (props.value === undefined) {
+      inputRef.current.value = null;
```

Comment:
Do we have the same warnings in JSDOM?

Can we try spying on `console.warn` and see if it is invoked? I just want to make sure we don't re-introduce this bug again in the future.

If not, it's cool. I do want to give it a try though.

Reply:
I cannot get a failing test to work on master.
```
    it('does not warn when clearing the input', () => {
      const warnSpy = jest.spyOn(window.console, 'warn');
      render(<Number {...requiredProps} />);
      userEvent.type(screen.getByLabelText('Test Component'), '1');
      fireEvent.keyDown(screen.getByLabelText('Test Component'), { key: 'Backspace' });
      expect(warnSpy.mock.calls.length).toEqual(0);
    });
```

Reply:
I am also not sure there is a *ton* of value having this regression test in the test suite 🤷